src/utils/index.ts

```diff
@@ -142,23 +142,37 @@ export function formatString(str: string, params?: object): string {
  * Generates the update mask for the provided object.
  * Note this will ignore the last key with value undefined.
  *
- * @param {[key: string]: any} obj The object to generate the update mask for.
- * @return {Array<string>} The computed update mask list.
+ * @param obj The object to generate the update mask for.
+ * @param maxPaths The optional map of keys for maximum paths to traverse.
+ *      Nested objects beyond that path will be ignored. This is useful for
+ *      keys with variable object values.
+ * @param currentPath The path so far.
+ * @return The computed update mask list.
  */
-export function generateUpdateMask(obj: {[key: string]: any}): string[] {
+export function generateUpdateMask(
+  obj: any, maxPaths: {[key: string]: boolean} = {}, currentPath = ''
```

Comment:
Rename currentPath to initialPath or root?

Reply:
Changed to `root`